runtime/lua/vim/shared.lua

```diff
@@ -174,9 +174,19 @@ function vim.tbl_extend(behavior, ...)
   if (behavior ~= 'error' and behavior ~= 'keep' and behavior ~= 'force') then
     error('invalid "behavior": '..tostring(behavior))
   end
+
+  if select('#', ...) < 2 then
+    error('wrong number of arguments (given '..tostring(1 + select('#', ...))..', expected at least 3)')
+  end
+
   local ret = {}
+  if getmetatable(select(1, ...)) == vim._empty_dict_mt then
```

Comment:
this should be written as `if vim and getmetatable(select(1, ...)) == vim._empty_dict_mt then`